### ISSUE TITLE
test-execute fails when CONFIG_IPV6_SIT is enabled

### ISSUE THREAD

**Issue by jku** [systemd/systemd#2852]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)
### systemd version the issue has been seen with

> 229
### Used distribution

> Yocto master (core-image-minimal with systemd as init manager)
### In case of bug report: Expected behaviour you didn't see

> test-execute should succeed
### In case of bug report: Unexpected behaviour you saw

> test-execute fails the test_exec_privatenetwork test because the service file uses this to verify that "PrivateNetwork=yes" worked:
> 
> ```
> ExecStart=/bin/sh -x -c 'i=$$(ip link | grep ": " | grep -v ": lo:"); test -z "$$i"'
> ```
> 
> the above fails if sit support (6to4 tunnel) is enabled in the kernel as "ip link" includes this even with PrivateNetwork=yes:
> 
> ```
> 2: sit0@NONE: <NOARP> mtu 1480 qdisc noop 
>   link/sit 0.0.0.0 brd 0.0.0.0
> ```
> 
> I suppose the test should succeed in this case.
### In case of bug report: Steps to reproduce the problem

> 1. Run kernel with CONFIG_IPV6_SIT=y (I'm sure a module works as well)
> 2. run test-execute

**Comment by evverx** [comment]
`test_exec_privatenetwork` tests:

> PrivateNetwork=
> Takes a boolean argument. If true, sets up a new network namespace for the executed processes and configures only the loopback network device "lo" inside it. No other network devices will be available to the executed process.
> https://www.freedesktop.org/software/systemd/man/systemd.exec.html#PrivateNetwork=

Hm...

```
# modprobe sit
# unshare --net ip link
1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN mode DEFAULT group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT group default qlen 1
    link/sit 0.0.0.0 brd 0.0.0.0

# modprobe --remove sit
# unshare --net ip link
1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN mode DEFAULT group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
```

**Comment by poettering** [comment]
Should be fixed by 6b08180ca6f1ceb913f6a69ffcaf96e9818fbdf5
